xstream/server/server/ruleManager.go

```diff
@@ -185,10 +186,27 @@ func stopRule(name string) (result string) {
 	} else {
 		result = fmt.Sprintf("Rule %s was not found.", name)
 	}
+	ruleProcessor.ExecReplaceRuleState(name, 1)
 	return
 }
 
 func restartRule(name string) error {
 	stopRule(name)
 	return startRule(name)
 }
+
+func recoverRule(name string) string {
+	rule, err := ruleProcessor.GetRuleByName(name)
+	if nil != err {
+		return fmt.Sprintf("%v", err)
+	}
+
+	err = startRule(name)
+	if nil != err {
+		return fmt.Sprintf("%v", err)
+	}
+	if 1 == rule.State {
```

Comment:
Can we check the rule state before starting it? The stopped rule should not be started at all